lib/interval.js

```diff
@@ -70,14 +70,19 @@ Interval.prototype.assign = function (lo, hi) {
   if (typeof lo !== 'number' || typeof hi !== 'number') {
     throw TypeError('Interval#assign: arguments must be numbers')
   }
-  if (isNaN(lo) || isNaN(hi) || lo > hi) {
+  //if (isNaN(lo) || isNaN(hi) || lo > hi) {
+  if (isNaN(lo) || isNaN(hi)) {
     return this.setEmpty()
   }
+  if (lo > hi){ return this.set(hi, lo)} // turn [a,b] to [b,a] if a > b.
```

Comment:
If the constructor creates an empty interval when `lo > hi` has actually some advantages when thinking on corner cases, for example consider the case described in `Interval.difference`

```js
var x = Interval(0, 1)
var n = Interval.difference(x, x) // n = Interval(next(1), 1) = EMPTY
isEmpty(n) // true
```